util/personal_names.py

```diff
@@ -32,27 +32,54 @@ def is_corporate_name(display_name):
     return False
 
 
-def display_name_to_sort_name(display_name):
+def display_name_to_sort_name(display_name, advanced=False):
```

Comment:
When would you use advanced=False? Every time you call this method in this branch or the metadata branch you pass in advanced=True. Looking at the code, it looks like you've improved the algorithm without adding latency. If you're confident in the new algorithm I would just use it to replace the old one.